fix: previous `texts` are still in memory by changing `texts` property

sectore opened the pull request:
Fixed by ~~clearing `textDisplayed` and~~ aborting Promise of `blink` whenever `texts` property has been changed.

### Before

[before.webm](https://github.com/satohshi/svelte-typewriter/assets/47693/4cac9016-239d-473d-a605-0353b8db3884)


### Now

[now.webm](https://github.com/satohshi/svelte-typewriter/assets/47693/d51e5697-a6bc-4c9a-a845-7e1062021436)

sectore committed:
fix: prev. `texts` are still in memory

sectore committed:
changeset

sectore committed:
Revert clearing `textDisplayed` - not needed

sectore committed:
Remove callback by running it

satohshi reviewed (commented):
I can't check without repro, but can't it be fixed with something like this?
```ts
let blinking: Animation

$effect(() => {
	typewriter(texts, repeat).catch((e: unknown) => {
		if (e instanceof DOMException && e.name === 'AbortError') return
		console.error(e)
	})
	return () => {
		blinking?.cancel()
		clearTimeout(timeout)
	}
})

const blink = (iterations: number) => {
	blinking = caret.animate([{ opacity: 0 }, { opacity: 1 }, { opacity: 0 }], {
		iterations,
		duration: blinkDuration,
		delay: blinkDuration / 4,
		easing: 'steps(2)',
	})
	return blinking.finished
}
```

Changing `texts` halfway seems to me like a pretty rare use case, so I'd rather not overcomplicate the code when wrapping the component in `{#key texts} {/key}` can do the job.

sectore committed:
simplify aborting of blink animation

as suggested by @satoshi https://github.com/satohshi/svelte-typewriter/pull/15#pullrequestreview-2130059905

sectore commented:
> I can't check without repro, but can't it be fixed with something like this?

Yes, it's simpler and it works fine - thx! _I did not know about `Animation.cancel`._

Fix 25d9e46 .

sectore committed:
pnpm format && lint

satohshi commented on src/lib/TypeWriter.svelte:
```diff
@@ -52,12 +56,13 @@
 	}
 
 	const blink = async (iterations: number) => {
```
It was an oversight on my part when I first made the change, but this function can, and should be synchronous
```suggestion
	const blink = (iterations: number) => {
```

satohshi commented on src/lib/TypeWriter.svelte:
```diff
@@ -52,12 +56,13 @@
 
 	const blink = async (iterations: number) => {
-		await caret.animate([{ opacity: 0 }, { opacity: 1 }, { opacity: 0 }], {
+		blinking = await caret.animate([{ opacity: 0 }, { opacity: 1 }, { opacity: 0 }], {
```
```suggestion
		blinking = caret.animate([{ opacity: 0 }, { opacity: 1 }, { opacity: 0 }], {
```

satohshi commented on src/lib/TypeWriter.svelte:
```diff
@@ -39,10 +39,14 @@
 	$effect(() => {
-		typewriter(texts, repeat)
-		return () => clearTimeout(timeout)
+		typewriter(texts, repeat).catch((_) => {})
```
I'd add this handling function to make future debugging easier if something else comes up
```suggestion
typewriter(texts, repeat).catch((e: unknown) => {
	if (e instanceof DOMException && e.name === 'AbortError') return
	console.error(e)
})
```

sectore committed:
Address change requests

satohshi reviewed (approved):
lgtm
